packages/react-charting/src/utilities/utilities.ts

```diff
@@ -1370,26 +1371,100 @@ export function wrapTextInsideDonut(selectorClass: string, maxWidth: number) {
       .attr('y', y)
       .attr('dy', lineNumber++ * lineHeight + 'em');
 
+    // Determine the length of a line
+    const lineLength = tspan.node()!.getBoundingClientRect().y;
+
+    // Determine the ellipsis length for word truncation
+    const ellipsis = '...';
+    tspan.text(ellipsis);
+    const ellipsisLength = tspan.node()!.getComputedTextLength();
+    tspan.text(null);
     while ((word = words.pop()!)) {
       line.push(word);
       tspan.text(line.join(' ') + ' ');
-      if (tspan.node()!.getComputedTextLength() > maxWidth && line.length > 1) {
+      // Determine if wrapping is required
+      // If yes, append a new line only if it does not exceed the max height
+      // here, max width = max height, so maxWidth is reused to check max height
```

Comment:
Does max height account for the curved shape. The available space for top line/bottom line is less than the middle line.
Or is this scenario too complex to handle.